sdk/core/azure-core/inc/azure/core/internal/io/parallel_file_body_stream.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief An internal implementation of Azure::IO::BodyStream that supports reading files in
+ * parallel.
+ *
+ */
+#pragma once
+
+#include "azure/core/io/body_stream.hpp"
+
+namespace Azure { namespace IO { namespace Internal {
+
+  /**
+   * @brief #Azure::IO::BodyStream providing its data from a file used for reading in parallel.
+   */
+  class ParallelFileBodyStream : public Azure::IO::BodyStream {
```

Comment:
This is a `FileBodyStream` with a starting offset in it. Itself doesn't have anything to do with parallelism. So I don't know if it's appropriate to call it `ParallelFileBodyStream`.

Reply:
Do you have a suggestion for a different name? I started off with PlatformFileBodyStream (or OSFileBodyStream), but open to alternatives.

Reply:
RandomAccessBodyStream

Reply:
I like it, should it include the word File? `RandomAccessFileBodyStream`?

Reply:
Yes, I like RandomAccessFileBodyStream